contracts/sfc/NodeDriverAuth.sol

```diff
@@ -182,12 +148,10 @@ contract NodeDriverAuth is OwnableUpgradeable, UUPSUpgradeable {
         sfc.sealEpochValidators(nextValidatorIDs);
     }
 
-    function isContract(address account) internal view returns (bool) {
-        return account.code.length > 0;
-    }
-
-    function _getCodeHash(address addr) internal view returns (bytes32) {
-        return addr.codehash;
+    /// Check the account is an EOA - it has no code or it has an EIP-7702 delegation
+    function isExternalAccount(address account) private returns (bool) {
+        bytes memory code = account.code;
+        return code.length == 0 || (code.length > 3 && code[0] == 0xef && code[1] == 0x01 && code[2] == 0x00);
```

Comment:
Fyi: a valid delegate indicator has to be 23 bytes long. Anything longer would not  be considered a delegate indicator, as shown for instance in the go-ethereum implementation [here](https://github.com/0xsoniclabs/go-ethereum/blob/949ae6d396a5798262c0d228a8de0e3fa504e00c/core/types/tx_setcode.go#L32-L43).

We might thus consider refining this to
```suggestion
        return code.length == 0 || (code.length == 23 && code[0] == 0xef && code[1] == 0x01 && code[2] == 0x00);
```